### PR TITLE
Dl 2591

### PR DESCRIPTION
**Pull request by MarjanMubarok**
DL-2591

Create new calls to Mongo Cache in ATED backend.

### CODE REVIEW

File: app/controllers/PropertyDetailsValuesController.scala [hunk 1]
```diff
@@ -81,32 +82,82 @@ trait PropertyDetailsValuesController extends BackendController  {
...
-            case Some(x) => Ok("")
+            case Some(_) => Ok("")
             case None => BadRequest("Invalid Request")
           }
         }
       }
   }
 
-  def saveDraftPropertyDetailsNewBuild(atedRefNo: String, id: String) = Action.async(parse.json) {
+  //TODO refactor/remove the method below in a future ticket
```
Comment (paulanderson1234) [inline app/controllers/PropertyDetailsValuesController.scala]: What is the reason that this is to be removed in a future ticket?

File: app/services/PropertyDetailsValuesService.scala [hunk 2]
```diff
@@ -127,42 +133,154 @@ trait PropertyDetailsValuesService extends ReliefConstants {
       }
       Future.successful(updatedPropertyDetails)
     }
+
     cacheDraftPropertyDetails(atedRefNo, updatePropertyDetails)
   }
 
+  //TODO This need to be removed in the future
```
Comment (paulanderson1234) [inline app/services/PropertyDetailsValuesService.scala]: Again - is there a need to hold onto this code for some reason?

File: conf/app.routes [hunk 3]
```diff
@@ -17,7 +17,12 @@ POST  /ated/:atedRefNo/property-details/has-value-change/:id        controllers.
 POST  /ated/:atedRefNo/property-details/acquisition/:id             controllers.PropertyDetailsValuesControllerImpl.saveDraftPropertyDetailsAcquisition(atedRefNo: String, id: String)
 POST  /ated/:atedRefNo/property-details/revalued/:id                controllers.PropertyDetailsValuesControllerImpl.saveDraftPropertyDetailsRevalued(atedRefNo: String, id: String)
 POST  /ated/:atedRefNo/property-details/owned-before/:id            controllers.PropertyDetailsValuesControllerImpl.saveDraftPropertyDetailsOwnedBefore(atedRefNo: String, id: String)
-POST  /ated/:atedRefNo/property-details/new-build/:id               controllers.PropertyDetailsValuesControllerImpl.saveDraftPropertyDetailsNewBuild(atedRefNo: String, id: String)
+## remove the old route rename for the new ones in the frontend
```
Comment (paulanderson1234) [inline conf/app.routes]: Presumably this tag is no longer required

File: test/controllers/PropertyDetailsValuesControllerSpec.scala [hunk 4]
```diff
@@ -107,120 +97,184 @@ class PropertyDetailsValuesControllerSpec extends PlaySpec with OneServerPerSuit
...
+        val result = controller.saveDraftPropertyDetailsIsNewBuild(testAccountRef, id = "1").apply(fakeRequest)
+        status(result) must be(OK)
+      }
 
-        val testPropertyDetails = PropertyDetailsBuilder.getPropertyDetails("1", Some("testPostCode1"))
-        val updated = PropertyDetailsNewBuild()
-        when(mockPropertyDetailsService.cacheDraftPropertyDetailsNewBuild(ArgumentMatchers.eq(testAccountRef),
-          ArgumentMatchers.eq("1"), ArgumentMatchers.eq(updated))(any())).thenReturn(Future.successful(Some(testPropertyDetails)))
+      "respond with a BAD REQUEST when failing to save the IsNewBuildFlag" in new Setup {
+        val update = new PropertyDetailsIsNewBuild()
```
Comment (paulanderson1234) [inline test/controllers/PropertyDetailsValuesControllerSpec.scala]: Is this required as has been defined Line 147 above?

File: app/controllers/PropertyDetailsValuesController.scala (hunk 1 above)
Comment (MarjanMubarok) [reply app/controllers/PropertyDetailsValuesController.scala]: Removed

File: app/services/PropertyDetailsValuesService.scala (hunk 2 above)
Comment (MarjanMubarok) [reply app/services/PropertyDetailsValuesService.scala]: No.It has been removed.

File: conf/app.routes (hunk 3 above)
Comment (MarjanMubarok) [reply conf/app.routes]: Removed

**Pull request merged**